server/v2/commands.go

```diff
@@ -11,62 +11,166 @@ import (
 	"syscall"
 
 	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
 	"github.com/spf13/viper"
 
+	"cosmossdk.io/core/server"
 	"cosmossdk.io/core/transaction"
 	"cosmossdk.io/log"
 )
 
-// Execute executes the root command of an application.
-// It handles adding core CLI flags, specifically the logging flags.
-func Execute(rootCmd *cobra.Command, envPrefix, defaultHome string) error {
-	rootCmd.PersistentFlags().String(FlagLogLevel, "info", "The logging level (trace|debug|info|warn|error|fatal|panic|disabled or '*:<level>,<key>:<level>')")
-	rootCmd.PersistentFlags().String(FlagLogFormat, "plain", "The logging format (json|plain)")
-	rootCmd.PersistentFlags().Bool(FlagLogNoColor, false, "Disable colored logs")
-	rootCmd.PersistentFlags().StringP(FlagHome, "", defaultHome, "directory for config and data")
+type RootCmdBuilder struct {
+	DefaultHome string
+	EnvPrefix   string
 
+	use     string
+	fixture CommandFixture
+	vipr    *viper.Viper
+	logger  log.Logger
+}
+
+type CommandFixture interface {
+	Bootstrap(cmd *cobra.Command) (WritesConfig, error)
+	RootCommand(
+		rootCommand *cobra.Command,
+		subCommand *cobra.Command,
+		logger log.Logger,
+		config server.ConfigMap,
+	) (*cobra.Command, error)
+}
+
+func NewRootCmdBuilder(
```

Comment:
Couldn't we take a normal cobra command, and just "enhance" it? Like autocli does.
I am personally not used to create cobra commands like that. It feels weird to me.

Reply:
working on a refactor to a factory pattern instead